app/models/update_lock.rb

```diff
@@ -0,0 +1,25 @@
+class UpdateLock
+  attr_reader :lockable, :locked_at
+
+  def initialize(lockable)
+    @lockable = lockable
+
+    if lockable && !lockable.respond_to?(:transmitted_at)
+      raise OutOfOrderTransmissionError.new("#{lockable.class} must implement transmitted_at")
+    end
+  end
+
+  def check_availability!(transmitted_at)
+    raise MissingAttributeError, "transmitted_at is mandatory" unless transmitted_at
+    transmitted_at = Float(transmitted_at)
```

Comment:
I'm real worried about using floats across two systems (via JSON) when we're using them specifically for the tiny fractions of a second.